sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs

```diff
@@ -502,6 +502,64 @@ internal static SummarySentenceCollection ConvertToSummarySentenceCollection(Ext
 
         #endregion
 
+        #region Abstract Summary
+
+        internal static List<Summary> ConvertToSummaryList(List<AbstractiveSummary> summaries)
+            => summaries.Select((summary) => new Summary(summary)).ToList();
+
+        internal static SummaryCollection ConvertToSummaryCollection(AbstractiveSummarizationResultBaseDocumentsItem document)
+        {
+            return new SummaryCollection(
+                ConvertToSummaryList(document.Summaries.ToList()),
+                ConvertToWarnings(document.Warnings));
+        }
+
+        internal static AbstractSummaryResultCollection ConvertToAbstractSummaryResultCollection(
+            AbstractiveSummarizationResult results,
+            IDictionary<string, int> idToIndexMap)
+        {
+            List<AbstractSummaryResult> abstractSummaryResults = new(results.Documents.Count);
+
+            // Read errors.
+            foreach (InputError error in results.Errors)
+            {
+                abstractSummaryResults.Add(new AbstractSummaryResult(error.Id, ConvertToError(error.Error)));
+            }
+
+            // Read results.
+            foreach (AbstractiveSummarizationResultBaseDocumentsItem document in results.Documents)
+            {
+                abstractSummaryResults.Add(new AbstractSummaryResult(
+                    document.Id,
+                    document.Statistics ?? default,
+                    ConvertToSummaryCollection(document)));
+            }
+
+            abstractSummaryResults = abstractSummaryResults.OrderBy(result => idToIndexMap[result.Id]).ToList();
+
+            return new AbstractSummaryResultCollection(abstractSummaryResults, results.Statistics, results.ModelVersion);
+        }
+
+        internal static AbstractSummaryResultCollection ConvertToAbstractSummaryResultCollection(
+            AnalyzeTextJobState jobState,
+            IDictionary<string, int> idToIndexMap)
+        {
+            AnalyzeTextLROResult task = jobState.Tasks.Items[0];
+            if (task.Kind == AnalyzeTextLROResultsKind.AbstractiveSummarizationLROResults)
+            {
+                return ConvertToAbstractSummaryResultCollection((task as AbstractiveSummarizationLROResult).Results, idToIndexMap);
+            }
+
+            if (task.Kind == AnalyzeTextLROResultsKind.AbstractiveSummarizationLROResults)
+            {
+                return ConvertToAbstractSummaryResultCollection((task as AbstractiveSummarizationLROResult).Results, idToIndexMap);
+            }
+
+            throw new InvalidOperationException($"Invalid task executed. Expected a {nameof(AnalyzeTextLROResultsKind.AbstractiveSummarizationLROResults)} or {nameof(AnalyzeTextLROResultsKind.AbstractiveSummarizationLROResults)} but instead got {task.Kind}.");
```

Comment:
this error looks like it would output the same thing twice where it says "expected a X or X". am I reading it wrong?

Reply:
Good catch. Thanks!